packages/react-router/src/link.tsx

```diff
@@ -56,13 +58,16 @@ export type Split<TValue, TIncludeTrailingSlash = true> = TValue extends unknown
       : never
   : never
 
-export type ParsePathParams<T extends string> = keyof {
-  [K in Trim<Split<T>[number], '_'> as K extends `$${infer L}`
-    ? L extends ''
+export type ParsePathParams<
```

Comment:
This comment may not really belong here at this line, rather it's something I noticed about the link component when testing out the PR.

In the "kitchen-sink-file-based" example, using a `<Link>` component it accepts the following `to` values.
1. `/route-group`
2. `route-group`

`1` works as expected, but `2` then says it requires params.
```
Property 'params' is missing in type '{ to: "route-group"; }' but required in type 'MakeRequiredPathParams<Router<Route<any, "/", "/", string, "__root__", RootSearchSchema, RootSearchSchema, RootSearchSchema, RootSearchSchema, ... 10 more ..., { ...; }>, TrailingSlashOption, Record<...>, Record<...>>, string, "route-group">'.ts(2741)
```

Also, something else, I noticed is that the `from` property doesn't accept just `route-group`, only allows `/route-group` unlike the `to` property shown above.

Reply:
Yeah, its a separate issue I'm going to be fixing in a separate PR. Its a bug on main as well.